docs/LICENSE

```diff
@@ -0,0 +1,437 @@
+Attribution-NonCommercial-ShareAlike 4.0 International
```

Comment:
I don't think you can just re-license the current files under this license if it's supposed to cover all subdirs... Also not sure if we want to use noncommercial license in php-src repo. That might need some discussion on internals.

Reply:
I picked this mostly because this is what the PHP internals book uses. CC-BY-NC-SA requires releasing changes under the same license. I'm not a license specialist at all. Maybe @ramsey can help.

Reply:
It is considered a non free license from what I read in https://en.wikipedia.org/wiki/Creative_Commons_NonCommercial_license and  https://creativecommons.org/public-domain/freeworks/ . If that's the case I would prefer not to include it in php-src.  We definitely need an internals discussion about this and potentially a vote.

Reply:
I kind of understand why it was chosen for internals book but having that as part of php-src repo might be problematic - I don't really know but have a feeling that it would be better to not have a non free license here. 

I'm not sure if this actually need to copy the internals book. It can be independent and more oriented on internal API. The internals book could stay for more generic things mainly oriented for extension developers.

Reply:
I think we have three options, basically:

* Rewrite everything from scratch.
* Use CC BY 4.0 for original content (we already use CC BY 3.0 for user documentation), and and CC BY-NC-SA 4.0 for pages with content copied from the PHP internals book.
* Ask the PHP internals book to switch to CC BY 4.0. This might be legally difficult, because possibly every contributor might have to give their consent. Ben is going to forward some notes he gathered for switching the PHP license to me.

Reply:
I think the only viable solution is rewrite everything from scratch as I don't think we should have anything licensed under CC BY-NC-SA and re-licensing might be tricky. But that's just my opinion so some wider discussion internals would be good. Potentially the PR's that were created by specific people that are happy with re-licensing could be ported here as those people are still copyright holders.

Btw. I think the stuff that is already there should be under PHP license as the primary repo license so it cannot be relicensed without contributors consent IMHO.

As I said I don't really think we should duplicate completely the internals book except some really technical parts that could be potentially rewritten.

Reply:
> Potentially the PR's that were created by specific people that are happy with re-licensing could be ported here as those people are still copyright holders.

Good point. Authors of specific PRs will be easier to track down and ask consent from.

> Btw. I think the stuff that is already there should be under PHP license as the primary repo license so it cannot be relicensed without contributors consent IMHO.

Right. I'll move `docs/src/miscellaneous/old` out of this PR.

> As I said I don't really think we should duplicate completely the internals book except some really technical parts that could be potentially rewritten.

The plan is not to duplicate it. The PHP internals book has a large focus on extensions, while I think the new documentation should focus on how the engine works. Nonetheless, there's still a large overlap (data structures, how to build, tests, zend_alloc, etc).

Reply:
> Right. I'll move docs/src/miscellaneous/old out of this PR.

We can just leave and keep everything licenses under PHP license. What's the actual point of using CC BY 4.0 for this?

Reply:
The PHP License _is_ ultimately aimed at software, with clauses such as the following:

https://opensource.org/license/php-3-01/

> Redistributions of any form whatsoever must retain the following acknowledgment: “This product includes PHP software, freely available from “.

There's also this one:

> 4. Products derived from this software may not be called “PHP”, nor may “PHP” appear in their name, without prior written permission from group at php.net. You may indicate that your software works in conjunction with PHP by saying “Foo for PHP” instead of calling it “PHP Foo” or “phpfoo” 

Does that mean a 3rd party, printed book cannot be called "The PHP internals" if it uses content from the docs? I suppose they may always ask for permission.

But as mentioned, I not at all familiar with licensing, so this might be just fine.

Reply:
Ok, let's avoid the licensing issues for now by using only original content. If we keep the pool of authors small for now we can still switch to CC later on if needed. There are some parts from php-internals-book I'd really like to use, I'll ask for permission from the original authors instead.

`docs/` is still moved because I decided to switch to Sphinx/rst because formatting tables in md proved difficult already. Sphinx should be more flexible in general for the future.